libraries/hotstuff/chain_pacemaker.cpp

```diff
@@ -107,6 +107,23 @@ namespace eosio { namespace hotstuff {
    {
    }
 
+   void chain_pacemaker::register_bcast_functions(
+           std::function<void(const chain::hs_proposal_message&)> on_proposal_message,
+           std::function<void(const chain::hs_vote_message&)> on_vote_message,
+           std::function<void(const chain::hs_new_block_message&)> on_new_block_message,
+           std::function<void(const chain::hs_new_view_message&)> on_new_view_message
+   ) {
+      FC_ASSERT(on_proposal_message, "on_proposal_message must be provided");
+      FC_ASSERT(on_vote_message, "on_proposal_message must be provided");
+      FC_ASSERT(on_new_block_message, "on_proposal_message must be provided");
+      FC_ASSERT(on_new_view_message, "on_proposal_message must be provided");
+      std::lock_guard g( _hotstuff_global_mutex ); // not actually needed but doesn't hurt
+      bcast_proposal_message = std::move(on_proposal_message);
+      bcast_vote_message = std::move(on_vote_message);
+      bcast_new_block_message = std::move(on_new_block_message);
+      bcast_new_view_message = std::move(on_new_view_message);
+   }
```

Comment:
Instead of 4 different functions, I would prefer a single function `broadcast_hotstuff_message` which takes a
`std::variant<chain::hs_proposal_message, chain::hs_vote_message, chain::hs_new_block_message, chain::hs_new_view_message>`. 
I think it would be a cleaner interface. What do you think?

Reply:
Done